design/sales.md

```diff
@@ -45,3 +45,49 @@ the Reservations module, so that it survives node restarts.
     Datastore API:
       function put(key, value)
       function get(key): value
+
+Reserving storage space
+-----------------------
+
+When a request for storage is submitted on chain, the sales module decides
+whether or not it wants to act on it. If the requested duration and size match
+what the node wants to sell and the reward is sufficient, then the sales module
+will go through several steps to try and fill a slot in the request.
+
+First, it will select a slot from the request to fill. Then, it will reserve
+space in the Repo to ensure that it keeps enough space available to store the
```

Comment:
As mentioned elsewhere, we should reserve enough space a-priory, not once the availability has been matched, this is so that the node doesn't use this space for other purposes. Reserving a large chunk that is further sliced by the availabilities is most likely better than doing so per availability, but doing it per availability is also possible.